Add syncTimeBeforePrivateRequests option

Tindtily opened the pull request:
## Summary
Add forceSync option to decides whether to force time synchronization before sending private requests

Tindtily commented:
for this #235

tiagosiebler commented on src/util/BaseRestClient.ts:
```diff
@@ -115,6 +115,8 @@ export default abstract class BaseRestClient {
       /** Request parameter values are now URI encoded by default during signing. Set to false to override this behaviour. */
       encodeSerialisedValues: true,
+      /** Force sync time before send private request */
+      forceSync: false,
```
I'm OK with this kind of optional feature, though I'd like to give it a self explanatory name:
```suggestion
      timeSyncBeforeRequest: false,
```

tiagosiebler replied:
In addition to the nitpicky rename of this new property, can you also add the boolean to the typescript interface for the REST options object? This is necessary to make it available to anyone building in typescript:
https://github.com/tiagosiebler/bybit-api/blob/master/src/util/requestUtils.ts#L1-L43

Tindtily replied:
I changed the var name, thx for reviewing

tiagosiebler reviewed (changes requested):
Thanks for the PR! Minor requests, thanks!

tiagosiebler commented on src/util/BaseRestClient.ts:
```diff
@@ -199,7 +201,7 @@ export default abstract class BaseRestClient {
-    if (this.timeOffset === null) {
+    if (this.timeOffset === null || this.options.syncTimeBeforePrivateRequests) {
       await this.syncTime();
     }
```
On reviewing this again, take a look at the `syncTime` function further down (line 450):
```typescript
    if (!force && !this.options.enable_time_sync) {
      this.timeOffset = 0;
      return Promise.resolve(false);
    }
```

It will bypass time sync if enable_time_sync is still disabled, although you could force it by changing 205 to pass the boolean:
```typescript
    if (this.timeOffset === null || this.options.syncTimeBeforePrivateRequests) {
      await this.syncTime(this.options.syncTimeBeforePrivateRequests === true);
    }
```

Tindtily replied:
I didn’t pay attention to the internal implementation of the `syncTime` function. I just manually commented out lines 205 and 206 locally. 
```
  prepareSignParams(method, signMethod, params, isPublicApi) {
    return __awaiter(this, void 0, void 0, function* () {
      if (isPublicApi) {
        return {
          originalParams: params,
          paramsWithSign: params,
        };
      }
      if (!this.key || !this.secret) {
        throw new Error("Private endpoints require api and private keys set");
      }
      //if (this.timeOffset === null) {
      yield this.syncTime();
      //}
      return this.signRequest(params || {}, method, signMethod);
    });
  }
```

There have been no exceptions with request_time and server_time for a long time. If I follow the internal logic of `syncTime`, my changes will still result in time errors, it  is a strange phenomenon. It seems that the difference between my local time and the server time is within a correct range.

Tindtily replied:
updated.

In addition, do I need to add instructions in readme

tiagosiebler replied:
Hmm your description sounds to me like any adjustment of the offset is incorrect and causes issues because of one (out of many) bad request. So with this turned on (including the force:true in sync time()), you're seeing more frequent time errors? 

Because without that change (force:true), syncTime won't function unless enable_time_sync is true...could you do more testing to see if this feature makes sense & improves things?

Tindtily replied:
> Because without that change (force:true), syncTime won't function unless enable_time_sync is true...could you do more testing to see if this feature makes sense & improves things?

I added a log in BaseRestClient.js
```
      console.log("before:", this.timeOffset);
      // Build a request and handle signature process
      const options = yield this.buildRequest(
        method,
        requestUrl,
        params,
        isPublicApi
      );
      console.log("after: ", this.timeOffset);
```
And still the manually change in `prepareSignParams`:
```
  prepareSignParams(method, signMethod, params, isPublicApi) {
    return __awaiter(this, void 0, void 0, function* () {
      if (isPublicApi) {
        return {
          originalParams: params,
          paramsWithSign: params,
        };
      }
      if (!this.key || !this.secret) {
        throw new Error("Private endpoints require api and private keys set");
      }
      //if (this.timeOffset === null) {
      yield this.syncTime();
      //}
      return this.signRequest(params || {}, method, signMethod);
    });
  }
```
It print out 
```
before: null
after:  0
```
So it turns out I don't need an offset for my local time

Tindtily replied:
I have a mac and my friend have a windows 11
Time errors occur frequently in wsl in my friend's win(cause the wsl kernel has some bug may be, I tried some online methods, but to no avail), while windows and mac do not currently see frequent problems. However, if the bybit server has time issues, I need to force synchronization.

Tindtily committed:
Add syncTimeBeforePrivateRequest option

Add syncTimeBeforePrivateRequest option to determines whether to perform time synchronization before sending private requests

Tindtily commented:
I double-checked, and because `syncTime` internal implementation is complicated, I changed `timeOffset` separately

tiagosiebler reviewed (approved):
Sorry about the delay. I'm happy with this approach, it doesn't impact existing functionality or introduce any risks in this implementation. Thanks for spending time on it. Will release it shortly to npm.
